File: Playwright/Aiagent/Playwright/pages/LoginPage.ts
import { Page } from '@playwright/test';

export class LoginPage {
  constructor(private page: Page) {}

  async goto() {
    await this.page.goto('/login');
  }

  get usernameField() {
    return this.page.getByRole('textbox', { name: 'Username' });
  }

  get passwordField() {
    return this.page.getByRole('textbox', { name: 'Password' });
  }

  get loginButton() {
    return this.page.getByRole('button', { name: 'Login' });
  }

  get invalidUsernameMessage() {
    return this.page.locator('#flash').getByText('Your username is invalid!');
  }

  get invalidPasswordMessage() {
    return this.page.locator('#flash').getByText('Your password is invalid!');
  }

  get successMessage() {
    return this.page.locator('#flash').getByText('You logged into a secure area!');
  }

  async login(username: string, password: string) {
    await this.usernameField.fill(username);
    await this.passwordField.fill(password);
    await this.loginButton.click();
  }
}

File: Playwright/AgenticQEframework/Agentic-QE-Framework/src/test/java/pages/LoginPage.java
package pages;

import com.microsoft.playwright.Page;
import com.microsoft.playwright.options.AriaRole;

public class LoginPage {
    private final Page page;

    // 1. Locators (Using the 2026 standard: Role-based locators)
    public LoginPage(Page page) {
        this.page = page;
    }

    // 2. Actions
    public void navigate() {
        page.navigate("https://www.saucedemo.com/");
    }

    public void login(String user, String pass) {
        page.getByPlaceholder("Username").fill(user);
        page.getByPlaceholder("Password").fill(pass);
        page.getByRole(AriaRole.BUTTON, new Page.GetByRoleOptions().setName("Login")).click();
    }
}